src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -49,11 +50,78 @@ pub trait ControllablePneumaticValve: PneumaticValve {
 }
 
 pub trait PneumaticContainer {
+    const HEAT_CAPACITY_RATIO: f64 = 1.4;
+    const HEAT_TRANSFER_SPEED: f64 = 2.5;
+    const HEAT_TRANSFER_COEFF: f64 = 1e-2;
+
     fn pressure(&self) -> Pressure;
+    // TODO: remove?
     fn volume(&self) -> Volume; // Not the volume of gas, but the physical measurements
     fn temperature(&self) -> ThermodynamicTemperature;
-    fn change_fluid_amount(&mut self, fluid_amount: Volume);
+    fn mass(&self) -> Mass;
+    fn change_fluid_amount(
+        &mut self,
+        fluid_amount: Mass,
+        fluid_temperature: ThermodynamicTemperature,
+    );
     fn update_temperature(&mut self, temperature_change: TemperatureInterval);
+
+    fn get_mass_flow_for_target_pressure(&self, target_pressure: Pressure) -> Mass {
+        ((target_pressure.get::<pascal>() / self.pressure().get::<pascal>())
+            .powf(1. / Self::HEAT_CAPACITY_RATIO)
+            - 1.)
+            * self.mass()
+    }
+
+    fn get_mass_flow_for_equilibrium(&self, other: &impl PneumaticContainer) -> Mass {
+        let p1 = self
+            .pressure()
+            .get::<pascal>()
+            .powf(1. / Self::HEAT_CAPACITY_RATIO);
+        let p2 = other
+            .pressure()
+            .get::<pascal>()
+            .powf(1. / Self::HEAT_CAPACITY_RATIO);
+        let m1 = self.mass().get::<kilogram>();
+        let m2 = other.mass().get::<kilogram>();
+        Mass::new::<kilogram>(m1 * m2 * (p2 - p1) / (m2 * p1 + m1 * p2))
+    }
+
+    /// Transfer heat between two containers depending on the temperature difference
+    fn heat_conduction(
+        &mut self,
+        context: &UpdateContext,
+        other: &mut impl PneumaticContainer,
+        transfer_factor: Ratio,
+    ) {
+        other.update_temperature(self.heat_conduction_single(
+            context,
+            other.temperature(),
+            transfer_factor,
+        ));
+    }
+
+    /**
+     * Transfer heat to the container depending on the temperature difference.
+     * Returns the temperature difference which would be applied to the opposite container.
+     */
+    fn heat_conduction_single(
```

Comment:
I think it should not be the responsibility of a `PneumaticContainer` to compute the temperature changes to apply to this and the other container. This calculation should be performed on the connecting element, which is presumably the `PneumaticContainerConnector` or `PneumaticExhaust`.

In the future, this calculation can then be replaced by the computation of the energy exchange. The effect of an addition of heat to a container can then be handled by the container, but not the calculation of how much heat should be exchanged.

Reply:
Previously both `PneumaticContainerConnector` and `PneumaticExhaust` had an implementation of the heat conduction. I wanted to remove the code duplication by putting it here.
I could move it back or maybe a better solution would be to move it out of the trait into a separate function?

Reply:
Yes, this is true. I thought about this for a bit and I think I'm not too bothered by the duplication. I'd rather have a bit of duplication here and there than to abstract this away prematurely.

However, there are options I see to avoid it if we this becomes a problem in the future and we have more and more duplicate implementations of this.

For one, it might be possible to treat the entire environmental atmosphere as a `PneumaticContainer` and then reduce the `PneumaticExhaust` to a connector between any container and the atmosphere. In this case we only need one implementation of heat conduction on the connector. However, I'm not sure how feasible this is right now.

Secondly, we could move the computation to a macro or a top-level function. Whether this is how you should handle it this way in Rust, I don't know 😄

Reply:
Ok. Then I will move the code back to `PneumaticContainerConnector` and `PneumaticExhaust`.